web/tests/api-spec.js

```diff
@@ -29,34 +19,37 @@ describe('api service', function () {
         
     });
 
+    afterEach(function() {
+        this.$httpBackend.verifyNoOutstandingExpectation();
+        this.$httpBackend.verifyNoOutstandingRequest();
+    });
+
     describe('getList method', function () {
         
-        it('should return a promise with the data of available forms', function (done) {
+        it('should return a promise with the data of available forms', function () {
             this.api.getList().then((data) => {
                 expect(data[0].name).toBe('How have you been using the site?');
                 expect(data[0].url).toBe('http://localhost:8000/forms/1');
-                done();
             });
             
-            this.$httpBackend.flush();
             this.$rootScope.$digest();
+            this.$httpBackend.flush();
         });
         
     });
 
     describe('getForm', function () {
 
-        it('should return a promise with the data of a form', function (done) {
+        it('should return a promise with the data of a form', function () {
             const url = 'http://localhost:8000/forms/1';
-            this.api.getForm(url).then((data) => {
 
+            this.api.getForm(url).then((data) => {
                 expect(data.name).toBe('How have you been using the site?');
                 expect(data.fieldsets[0].name).toBe('Free text field');
-            
-                done();
             });
-            this.$httpBackend.flush();
+
             this.$rootScope.$digest();
+            this.$httpBackend.flush();
```

Comment:
Would it be useful for readability to have `this.resolvePromises()` that calls this, so that the end of the test describes why it needs to do that?
```js
beforeEach(function () {
    this.resolvePromises = function () {
        this.$rootScope.$digest();
        this.$httpBackend.flush();
    };
});

it('should get some data', function () {
    this.api.getSomething().then((data) => {
        expect(data).toBeData();
    });
    this.resolvePromises();
});
```

Or `resolvePromises` could be `endHttpSpec`